File: frontend/lib/models/user_details/activity_level.dart
enum ActivityLevel {
  veryLow(0, 'very_low'),
  light(1, 'light'),
  moderate(2, 'moderate'),
  active(3, 'active'),
  veryActive(4, 'very_active');

  final int value;
  final String nameStr;

  const ActivityLevel(this.value, this.nameStr);

  String toJson() => nameStr;

  static ActivityLevel fromJson(String value) {
    return ActivityLevel.values.firstWhere(
      (e) => e.nameStr == value.toUpperCase(),
      orElse: () => throw ArgumentError('Unknown activity level: $value'),
    );
  }

  int toInt() => value;

  static ActivityLevel fromInt(int value) {
    return ActivityLevel.values.firstWhere(
      (e) => e.value == value,
      orElse: () => throw ArgumentError('Invalid activity level value: $value'),
    );
  }
}

File: backend/user_details/enums/activity_level.py
from enum import Enum


class ActivityLevel(str, Enum):
    VERY_LOW = "very_low"
    LIGHT = "light"
    MODERATE = "moderate"
    ACTIVE = "active"
    VERY_ACTIVE = "very_active"

    def get_pal(self):
        return {
            ActivityLevel.VERY_LOW: 1.2,
            ActivityLevel.LIGHT: 1.4,
            ActivityLevel.MODERATE: 1.6,
            ActivityLevel.ACTIVE: 1.8,
            ActivityLevel.VERY_ACTIVE: 2.0,
        }[self]